File: gorouter-compiler/src/main/java/cn/gorouter/gorouter_compiler/processor/BaseProcessor.java
package cn.gorouter.gorouter_compiler.processor;

import java.util.Set;

import javax.annotation.processing.AbstractProcessor;
import javax.annotation.processing.Filer;
import javax.annotation.processing.ProcessingEnvironment;
import javax.annotation.processing.RoundEnvironment;
import javax.lang.model.element.TypeElement;
import javax.lang.model.util.Elements;
import javax.lang.model.util.Types;

import cn.gorouter.gorouter_compiler.utils.TypeUtils;

public class BaseProcessor extends AbstractProcessor {

    private Filer mFiler;
    private Types types;
    private Elements elementUtils;
    private TypeUtils typeUtils;

    @Override
    public synchronized void init(ProcessingEnvironment processingEnv) {
        super.init(processingEnv);

        mFiler = processingEnv.getFiler();
        types = processingEnv.getTypeUtils();
        elementUtils = processingEnv.getElementUtils();
        typeUtils = new TypeUtils(types , elementUtils);
    }

    @Override
    public boolean process(Set<? extends TypeElement> set, RoundEnvironment roundEnvironment) {
        return false;
    }
}

File: gorouter-compiler-kotlin/src/main/java/cn/gorouter/compiler/kotlin/BaseProcessor.kt
package cn.gorouter.compiler.kotlin

import javax.annotation.processing.AbstractProcessor
import javax.annotation.processing.ProcessingEnvironment
import javax.annotation.processing.RoundEnvironment
import javax.lang.model.element.TypeElement

abstract class BaseProcessor : AbstractProcessor() {
    protected var logger : Logger ?= null

    override fun init(processingEnvironment: ProcessingEnvironment?) {
        super.init(processingEnvironment)
        processingEnvironment?.messager?.let {
            logger = Logger(it)
        }

    }
    override fun process(annotations: MutableSet<out TypeElement>?, roundEnv: RoundEnvironment?): Boolean {

        return false
    }

    protected abstract fun processRoute(nodeMap: MutableMap<String , String>)

    protected abstract fun writeToCache(nodeMap: MutableMap<String , String>)
}